### PR TITLE
Cobbler: automatically run ansible after install of new-style profiles

### PR DESCRIPTION
**Pull request by zmc**
For now it only uses ``--tags user,pubkeys``, because I'm not sure if running the whole thing is safe. It might be distro dependant and is definitely something we'd want to test carefully. This should be safe though.

### CODE REVIEW

File: roles/cobbler/templates/triggers/install/post/cephlab_ansible.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+## {{ ansible_managed }}
+set -ex
+name=$2
+export USER=root
+export HOME=/root
+ANSIBLE_CM_PATH=/root/ceph-cm-ansible
+
+# Bail if the ssh port isn't open.
+nc -vz $name 22
```
Comment (andrewschoen) [inline roles/cobbler/templates/triggers/install/post/cephlab_ansible.sh]: Can we leave some comments about how this is meant to fail silently here the first time it's ran, but then is called again by the rc.local after first reboot?

File: roles/cobbler/templates/triggers/install/post/cephlab_ansible.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+mkdir -p /var/log/ansible
+
+pushd $ANSIBLE_CM_PATH
+ANSIBLE_SSH_PIPELINING=1 ansible-playbook testnodes.yml -vv --limit $name* --tags user,pubkeys 2>&1 | tee /var/log/ansible/$name.log
```
Comment (andrewschoen) [inline roles/cobbler/templates/triggers/install/post/cephlab_ansible.sh]: Once we merge https://github.com/ceph/ceph-cm-ansible/pull/1 I think the users.yml playbook would be perfect to use here.  It essentially does what --tags user,pubkeys is doing but will work all for all roles.

**Comment by ktdreyer** [comment]
Just a nit, but it's nice to prefix cobbler-specific commits and PRs with "cobbler:" in the future.

**Comment by zmc** [comment]
Good point @ktdreyer

**Comment by andrewschoen** [comment]
@ktdreyer maybe we should have a standard of 'role: message' for all commits in this repo?

**Comment by ktdreyer** [comment]
definitely

File: roles/cobbler/templates/snippets/cephlab_rc_local [hunk 3]
```diff
@@ -0,0 +1,24 @@
+## {{ ansible_managed }}
+#set lockfile = '/.cephlab_rc_local'
+#set script = '/etc/rc.local'
+
+cat > $script << EOF
+#!/bin/bash
```
Comment (andrewschoen) [inline roles/cobbler/templates/snippets/cephlab_rc_local]: should this be #!/usr/bin/env bash ?

File: roles/cobbler/templates/triggers/install/post/cephlab_ansible.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```
Comment (andrewschoen) [inline roles/cobbler/templates/triggers/install/post/cephlab_ansible.sh]: should this also be #!/usr/bin/env bash ?

Comment (ktdreyer) [inline roles/cobbler/templates/triggers/install/post/cephlab_ansible.sh]: /bin/bash is fine IMHO

Comment (zmc) [inline roles/cobbler/templates/triggers/install/post/cephlab_ansible.sh]: Yeah, /bin/bash works on RHEL 7 and Ubuntu 14.04; if it starts failing on older stuff I'll figure something out.

**Pull request merged**